tests/components/growatt/test_sensor.py

```diff
@@ -0,0 +1,130 @@
+"""The tests for the Growatt platform."""
+import json
+from unittest import mock
+
+import pytest
+import requests_mock
+import growatt
+
+import homeassistant.components.growatt.sensor as victim
+
+
+@mock.patch("growatt.GrowattApi.login", return_value={"userId": "1"})
+def test_login(_):
+    """Test logging in."""
+    client = growatt.GrowattApi()
+    login_success = victim.login(client, "foo", "bar")
+    assert login_success
+
+
+def test_convert_multiplier():
+    """Test converting multipliers."""
+    assert (
+        victim.GrowattPlant(None, None, "", "").convert_multiplier(
+            "kg", {"g": 1, "kg": 1000}
+        )
+        == 1000
+    )
+
+
+def test_convert_multiplier_no_value():
+    """Test converting multipliers with exception."""
+    with pytest.raises(ValueError):
+        victim.GrowattPlant(None, None, "", "").convert_multiplier(
+            "kg", {"g": 1, "mg": 0.001}
+        )
+
+
+def test_convert_to_kwh():
+    """Test converting to kWh."""
+    assert (
+        victim.GrowattPlantTotals(None, None, "", "")._convert_to_kwh(
+            "5.42", "GWh"
+        )
+        == 5420000
+    )
+
+
+@mock.patch(
+    "growatt.GrowattApi.plant_detail", return_value={"data": "some-data"}
+)
+def test_plant_list(_):
+    """Test getting the list of plants."""
+    sensor = victim.GrowattPlant(None, growatt.GrowattApi(), "foo", "bar")
+    login_res = sensor._client.plant_detail("1")
```

Comment:
It seems only test upstream lib, we don't need it.

Reply:
Removed